screenpy/actions/silently.py

```diff
@@ -157,11 +47,46 @@ def Silently(duck: T_duck) -> Union[T_duck, T_silent_duck]:
     if settings.UNABRIDGED_NARRATION:
         return duck
 
+    # mypy really doesn't like monkeypatching
+    # See https://github.com/python/mypy/issues/2427
+
     if isinstance(duck, Performable):
-        return SilentlyPerformable(duck)
+        original_perform_as = duck.perform_as
+
+        def perform_as(self: Performable, actor: Actor) -> None:  # noqa: ARG001
+            """Direct the Actor to perform silently."""
+            with the_narrator.mic_cable_kinked():
+                original_perform_as(actor)
+                if not settings.UNABRIDGED_NARRATION:
+                    the_narrator.clear_backup()
+                return
+
+        duck.perform_as = types.MethodType(perform_as, duck)  # type: ignore[method-assign]
+
     if isinstance(duck, Answerable):
-        return SilentlyAnswerable(duck)
+        original_answered_by = duck.answered_by
+
+        def answered_by(self: Answerable, actor: Actor) -> Any:  # noqa: ARG001
+            """Direct the Actor to answer the question silently."""
+            with the_narrator.mic_cable_kinked():
+                thing = original_answered_by(actor)
+                if not settings.UNABRIDGED_NARRATION:
+                    the_narrator.clear_backup()
+                return thing
+
+        duck.answered_by = types.MethodType(answered_by, duck)  # type: ignore[method-assign]
+
     if isinstance(duck, Resolvable):
-        return SilentlyResolvable(duck)
+        original_resolve = duck.resolve
+
+        def resolve(self: Resolvable) -> Matcher:  # noqa: ARG001
+            """Produce the Matcher to make the assertion, silently."""
+            with the_narrator.mic_cable_kinked():
+                res = original_resolve()
+                if not settings.UNABRIDGED_NARRATION:
+                    the_narrator.clear_backup()
+                return res
+
+        duck.resolve = types.MethodType(resolve, duck)  # type: ignore[method-assign]
```

Comment:
What is the purpose of this `types.MethodType`? What does it do?

Reply:
Allows you to bind `resolve` to the instance of `duck`.  Otherwise it binds it to the class only.

Reply:
How did you find this library? 

Can you not just `duck.resolve = resolve`? Does that lead to `isinstance` not working?